lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/file/download/v2/DownloadListDaoImplTest.java

```diff
@@ -322,17 +352,724 @@ public void testClearDownloadListWithMultipleUsers() {
+		assertFalse(DownloadListDAOImpl.isMatch(item, result));
+	}
+
+	@Test
+	public void testIsMatchWithVersionNullResult() {
+		DownloadListItem item = new DownloadListItem().setFileEntityId("syn123").setVersionNumber(1L);
+		DownloadListItemResult result = (DownloadListItemResult) new DownloadListItemResult().setFileEntityId("syn123")
+				.setVersionNumber(null);
+		// call under test
+		assertFalse(DownloadListDAOImpl.isMatch(item, result));
+	}
 	
+	@Test
+	public void testReadTempoaryTableOfAvailableFilesWithEmptyDownloadLists() {
+		int batchSize = 100;
+		// deny access to all files
+		// Call under test
+		List<Long> results = downloadListDao.readTempoaryTableOfAvailableFiles(l -> Collections.emptyList(),
+				userOneIdLong, batchSize);
+		assertEquals(Collections.emptyList(), results);
+	}
+
+	@Test
+	public void testReadTempoaryTableOfAvailableFilesWithNoDownloadAccess() {
+		int numberOfProject = 1;
+		int foldersPerProject = 1;
+		int filesPerFolder = 5;
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+
+		// Add the latest version of each file to the user's download list
+		List<DownloadListItem> toAdd = files.stream()
+				.map(n -> new DownloadListItem().setFileEntityId(n.getId()).setVersionNumber(null))
+				.collect(Collectors.toList());
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, toAdd);
+		// Add the first version of each file to the user's download list
+		toAdd = files.stream().map(n -> new DownloadListItem().setFileEntityId(n.getId()).setVersionNumber(1L))
+				.collect(Collectors.toList());
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, toAdd);
+
+		int batchSize = 100;
+		// deny access to all files
+		// Call under test
+		List<Long> results = downloadListDao.readTempoaryTableOfAvailableFiles(l -> Collections.emptyList(),
+				userOneIdLong, batchSize);
+		assertEquals(Collections.emptyList(), results);
+	}
+
+	@Test
+	public void testReadTempoaryTableOfAvailableFilesWithFullAccess() {
+		int numberOfProject = 1;
+		int foldersPerProject = 1;
+		int filesPerFolder = 5;
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+
+		// Add the latest version of each file to the user's download list
+		List<DownloadListItem> toAdd = files.stream()
+				.map(n -> new DownloadListItem().setFileEntityId(n.getId()).setVersionNumber(null))
+				.collect(Collectors.toList());
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, toAdd);
+		// Add the first version of each file to the user's download list
+		toAdd = files.stream().map(n -> new DownloadListItem().setFileEntityId(n.getId()).setVersionNumber(1L))
+				.collect(Collectors.toList());
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, toAdd);
+
+		int batchSize = 100;
+		// grant access to all files
+		// Call under test
+		List<Long> results = downloadListDao.readTempoaryTableOfAvailableFiles(l -> l, userOneIdLong, batchSize);
```

Comment:
l -> l can done using Function::identity

Reply:
I cannot get this to compile.   I think I would need to redefine EntityAccessCallback to be a generic Function<List<Long>, List<Long>>

Reply:
Right, you could implement the interface but not worth it.